ui-admin/src/study/kits/kitcollection/KitScanner.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { ReactNode, useState } from 'react'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import { Button } from 'components/forms/Button'
+import { doApiLoad } from 'api/api-utils'
+import Api from 'api/api'
+import { dateToDefaultString, Enrollee } from '@juniper/ui-core'
+import { TextInput } from 'components/forms/TextInput'
+import { renderPageHeader } from 'util/pageUtils'
+import InfoPopup from 'components/forms/InfoPopup'
+import { useUser } from 'user/UserProvider'
+import { BarcodeScanner } from './BarcodeScanner'
+import Select from 'react-select'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import {
+  faCircleCheck,
+  faCircleDot,
+  faCircleExclamation,
+  faCircleQuestion,
+  faSearch
+} from '@fortawesome/free-solid-svg-icons'
+import { Checkbox } from 'components/forms/Checkbox'
+
+const kitScanModeOptions = [
+  { value: 'ASSIGN', label: 'Assign a new kit' },
+  { value: 'COLLECT', label: 'Collect a completed kit' }
+]
+
+export const KitScanner = ({ studyEnvContext }: { studyEnvContext: StudyEnvContextT }) => {
+  const [enableManualOverride, setEnableManualOverride] = useState(false)
+  const [selectedScanMode, setSelectedScanMode] = useState<{ value: string, label: string } | undefined>()
+
+  //Step 1 state
+  const [showEnrolleeCodeScanner, setShowEnrolleeCodeScanner] = useState(false)
+  const [enrolleeShortcode, setEnrolleeShortcode] = useState<string>()
+  const [enrollee, setEnrollee] = useState<Enrollee>()
+  const [enrolleeCodeError, setEnrolleeCodeError] = useState<string>()
+
+  //Step 2 state
+  const [isEnrolleeIdentityConfirmed, setIsEnrolleeIdentityConfirmed] = useState(false)
+
+  //Step 3 state
+  const [showKitScanner, setShowKitScanner] = useState(false)
+  const [kitBarcode, setKitBarcode] = useState<string | undefined>()
+  const [kitCodeError, setKitCodeError] = useState<string>()
+
+  const { user } = useUser()
+
+  const assignKit = async () => {
+    if (!enrollee || !kitBarcode) { return }
+    doApiLoad(async () => {
+      await Api.createKitRequest(
+        studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+          enrollee!.shortcode,
+          {
+            kitType: 'SALIVA',
```

Comment:
I'm hoping there's actually a way to determine the kit type from the barcode. I'd rather not have the admin select it here to prevent accidental mismatch because the admin isn't the source of truth for this stuff